judge/jinja2/submission.py

```diff
@@ -15,6 +16,8 @@ def submission_layout(submission, profile_id, user, completed_problem_ids, edita
     elif profile_id == submission.user_id:
         can_view = True
     elif submission.problem_id in completed_problem_ids:
-        can_view = submission.problem.is_public or submission.problem_id in tester_problem_ids
+        can_view = submission.problem_id in tester_problem_ids
+        if not private_submission:
```

Comment:
Instead of passing the `settings.DMOJ_PRIVATE_SUBMISSION` into the context and then re-checking, you should check it here directly instead.

Reply:
Thank you for the reply.
I did it directly.